content/blog/2017/2017-01-31-pipeline-scalability-best-practice.adoc

```diff
@@ -0,0 +1,197 @@
+---
+layout: post
+title: "Best Practices for Scalable Pipeline Code"
+tags:
+- pipeline
+- performance
+- scalability
+author: svanoort
+---
+====
+This by link:https://github.com/svanoort[Sam Van Oort], Software Engineer at
+link:http://cloudbees.com[CloudBees] and contributor to the Jenkins project.
+====
+
+=== Promises
+
+Today I'm going to show you best practices to write scalable and robust Jenkins pipeline. This is drawn from a
+combination of work with the internals of Pipeline and observations with high-scalability users.
+
+Pipeline code works beautifully for its intended role of orchestrating
+build/test/deploy/administer cycles.  As it is pressed into more complex roles and unanticipated uses, some users hit issues.
```

Comment:
"orchestrating `$X` cycles" is not verbiage that maps to any existing Pipeline documentation or blog posts. I get what you're trying to convey here, so perhaps "its intended role of executing build/test/deploy/administer tasks" 

That would map a little more consistently with the phrasing from the [Why Pipeline](https://jenkins.io/doc/book/pipeline/#why) part of the user handbook